### PR TITLE
[Feature][Restapi] Allow metrics information to be associated to logical plan nodes

### PR DESCRIPTION
**Pull request by liugddx**
### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/sink/SeaTunnelSink.java [hunk 1]
```diff
@@ -135,4 +136,13 @@ default Optional<Serializer<CommitInfoT>> getCommitInfoSerializer() {
     default Optional<Serializer<AggregatedCommitInfoT>> getAggregatedCommitInfoSerializer() {
         return Optional.empty();
     }
+
+    /**
+     * Get the catalog table of the sink.
+     *
+     * @return Optional of catalog table.
+     */
+    default Optional<CatalogTable> getWriteCatalogTable() {
```
Comment (Hisoka-X) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/sink/SeaTunnelSink.java:145]: cc @hailin0

File: seatunnel-engine/seatunnel-engine-core/src/main/java/org/apache/seatunnel/engine/core/job/VertexInfo.java [hunk 2]
```diff
@@ -35,4 +36,6 @@ public class VertexInfo implements Serializable {
     private PluginType type;
 
     private String connectorType;
+
+    private List<String> tablePaths;
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-core/src/main/java/org/apache/seatunnel/engine/core/job/VertexInfo.java]:
```suggestion
    private List<TablePath> tablePaths;
```

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/DAGUtils.java [hunk 3]
```diff
@@ -63,13 +71,57 @@ public static JobDAGInfo getJobDAGInfo(
                         pipeline.getVertexes()
                                 .forEach(
                                         (id, vertex) -> {
+                                            List<String> tablePaths = new ArrayList<>();
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/DAGUtils.java]: Please refactor the code. The same code show up twice.

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/sink/SeaTunnelSink.java (hunk 1 above)
Comment (Hisoka-X) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/sink/SeaTunnelSink.java:145]: Please add test case in https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/src/test/java/org/apache/seatunnel/api/connector/ConnectorSpecificationCheckTest.java#L50 to make sure all connector sink implement `getWriteCatalogTable`

Comment (liugddx) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/sink/SeaTunnelSink.java:145]: > Please add test case in https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/src/test/java/org/apache/seatunnel/api/connector/ConnectorSpecificationCheckTest.java#L50 to make sure all connector sink implement `getWriteCatalogTable`

Done.

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/JsonUtils.java [hunk 4]
```diff
@@ -277,12 +277,12 @@ public static ArrayNode parseArray(String text) {
...
     /** json serializer */
-    public static class JsonDataSerializer extends JsonSerializer<String> {
+    public static class JsonDataSerializer extends JsonSerializer<Object> {
 
         @Override
-        public void serialize(String value, JsonGenerator gen, SerializerProvider provider)
+        public void serialize(Object value, JsonGenerator gen, SerializerProvider provider)
                 throws IOException {
-            gen.writeRawValue(value);
+            gen.writeRawValue(value.toString());
```
Comment (Hisoka-X) [inline seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/JsonUtils.java]: Are we sure the value should not be null?

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/DAGUtils.java [hunk 5]
```diff
@@ -119,4 +130,38 @@ public static JobDAGInfo getJobDAGInfo(
...
+                List<CatalogTable> producedCatalogTables =
+                        sourceAction.getSource().getProducedCatalogTables();
+                List<TablePath> sourceTablePaths =
+                        producedCatalogTables.stream()
+                                .map(CatalogTable::getTablePath)
+                                .collect(Collectors.toList());
+                tablePaths.addAll(sourceTablePaths);
+            } catch (UnsupportedOperationException e) {
+                // ignore
+                tablePaths.add(TablePath.DEFAULT);
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/DAGUtils.java:157]: Add warning?

File: seatunnel-engine/seatunnel-engine-core/src/main/java/org/apache/seatunnel/engine/core/job/JobDAGInfo.java [hunk 6]
```diff
@@ -32,4 +39,35 @@ public class JobDAGInfo implements Serializable {
...
+            for (Edge edge : entry.getValue()) {
+                JsonObject edgeJsonObject = new JsonObject();
+                edgeJsonObject.add("inputVertexId", edge.getInputVertexId().toString());
+                edgeJsonObject.add("targetVertexId", edge.getTargetVertexId().toString());
+                jsonArray.add(edgeJsonObject);
+            }
+            pipelineEdgesJsonObject.add(entry.getKey().toString(), jsonArray);
+        }
+
+        ObjectMapper objectMapper = new ObjectMapper();
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-core/src/main/java/org/apache/seatunnel/engine/core/job/JobDAGInfo.java]: please use static ObjectMapper

Comment (liugddx) [reply seatunnel-engine/seatunnel-engine-core/src/main/java/org/apache/seatunnel/engine/core/job/JobDAGInfo.java]: Done.

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java [hunk 7]
```diff
@@ -752,7 +752,7 @@ private JsonObject getJobInfoJson(JobState jobState, String jobMetrics, JobDAGIn
                         DateTimeUtils.toString(
                                 jobState.getFinishTime(),
                                 DateTimeUtils.Formatter.YYYY_MM_DD_HH_MM_SS))
-                .add(RestConstant.JOB_DAG, JsonUtils.toJsonString(jobDAGInfo))
+                .add(RestConstant.JOB_DAG, jobDAGInfo.toJsonObject())
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java:757]: Seem like we need add test case to check job dag value?

Comment (liugddx) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java:757]: > Seem like we need add test case to check job dag value?

Done.

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java [hunk 8]
```diff
@@ -327,6 +329,38 @@ public void testGetJobInfoByJobId() {
...
+                                                        equalTo("1"))
+                                                .body(
+                                                        "jobDag.pipelineEdges['1'][0].targetVertexId",
+                                                        equalTo("2"))
+                                                .body("jobDag.vertexInfoMap", hasSize(2))
+                                                .body(
+                                                        "jobDag.vertexInfoMap[0].vertexId",
+                                                        equalTo(1))
+                                                .body(
+                                                        "jobDag.vertexInfoMap[0].type",
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java:350]: check for vertexName?

File: docs/en/seatunnel-engine/rest-api-v1.md [hunk 9]
```diff
@@ -217,10 +225,18 @@ This API has been deprecated, please use /hazelcast/rest/maps/job-info/:jobId in
   "jobStatus": "",
   "createTime": "",
   "jobDag": {
-    "vertices": [
+    "jobId": "",
+    "vertexInfoMap": [
+      {
+        "vertexId": 1,
+        "type": "",
+        "connectorType": "",
```
Comment (Hisoka-X) [inline docs/en/seatunnel-engine/rest-api-v1.md]: connectorType -> vertexName

Comment (liugddx) [reply docs/en/seatunnel-engine/rest-api-v1.md]: Done

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java (hunk 8 above)
Comment (liugddx) [reply seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java:350]: Done

**Review by Hisoka-X — DISMISSED** [review]

**Review by Hisoka-X — APPROVED** [review]

**Review by liunaijie — APPROVED** [review]
LGTM

**Pull request merged**